packages/flutter_tools/lib/src/base/pub_spec_content.dart

```diff
@@ -0,0 +1,36 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:yaml/yaml.dart';
+
+class PubContent {
+  PubContent(this.content);
```

Comment:
nit, this constructor can be const. for the real tool, we probably won't be able to make any instances const (since we will be reading the content from disk), but maybe in tests?

```suggestion
  const PubContent(this.content);
```